homeassistant/components/sonos/__init__.py

```diff
@@ -483,13 +483,14 @@ def async_discovered_player(
         if uid not in self.data.discovery_known:
             _LOGGER.debug("New %s discovery uid=%s: %s", source, uid, info)
             self.data.discovery_known.add(uid)
-        asyncio.create_task(
+        self.hass.async_create_background_task(
```

Comment:
Looks like this doesn't need to be a background task and we can just await this if we change the wrapping method to a coroutine function.

Code owners can potentially follow up with this refactor if confirmed.

Reply:
The chain of calls this triggers can take several seconds to complete and could impact startup. I think it should remain a backup task.

Reply:
It's enough to occasionally exceed the 5s limit in tests.

Reply:
Just so that I understand, what do you mean with "5s limit in tests"?

Reply:
Oh, my mistake. I've been using a 5s timeout in a local alias to run tests for so long I forgot it's not the project default.

Anyway, if this is _not_ a background task the full integration test suite takes ~100s and a couple individual tests will fail to complete in 5s. If a background task is used the full suite completes in <10s.

Perhaps there's a better way to design the integration setup and/or the tests, but a background task still seems appropriate for this call.

Reply:
It sounds like the tests are waiting for the `ZGS_SUBSCRIPTION_TIMEOUT * 2` if it's not a background task, and we'd need to patch that.

I think it's ok to keep it like a background task though for now.